src/server/pps/server/pipeline_controller.go

```diff
@@ -149,11 +150,9 @@ func (pc *pipelineController) Start(timestamp time.Time) {
 		case <-pc.ctx.Done():
 			return
 		case <-pc.bumpChan:
-			err := pc.step(timestamp)
-			// we've given up on the step, check if the error indicated that the pipeline should fail
-			if err != nil {
-				log.Errorf("PPS master: error starting a pipelineController for pipeline '%s': %v", pc.pipeline,
-					errors.Wrapf(err, "failing pipeline %q", pc.pipeline))
+			// pc.step returns true if the pipeline was deleted, and the controller can try to shutdown
+			if pc.step(timestamp) {
```

Comment:
I haven't read the surrounding code so this suggestion may make no sense, but: do we use `timestamp` outside of `pc.step()`? If not, would it make sense to instantiate it inside of `pc.step()`? (basically to limit its lifetime to the function it's used in)